src/main/java/com/dnd/gongmuin/auth/service/AuthService.java

```diff
@@ -181,4 +199,62 @@ private void updateAdditionalInfo(AdditionalInfoRequest request, Member findMemb
 		);
 	}
 
+	@Transactional
+	public DeleteMemberResponse deleteMember(HttpServletRequest request) {
+		String accessToken = cookieUtil.getCookieValue(request);
+
+		if (!tokenProvider.validateToken(accessToken, new Date())) {
+			throw new ValidationException(AuthErrorCode.UNAUTHORIZED_TOKEN);
+		}
+
+		Authentication authentication = tokenProvider.getAuthentication(accessToken);
+		Member member = (Member)authentication.getPrincipal();
+
+		// RefreshToken 삭제
+		if (!Objects.isNull(redisUtil.getValues("RT:" + member.getSocialEmail()))) {
+			redisUtil.deleteValues("RT:" + member.getSocialEmail());
+		}
+
+		// 현재 발급 되어 있는 AccessToken 블랙리스트 등록
+		Long expiration = tokenProvider.getExpiration(accessToken, new Date());
+		redisUtil.setValues(accessToken, DELETE, Duration.ofMillis(expiration));
+
+		// AccessToken 블랙리스트 등록 여부 검증
+		String values = redisUtil.getValues(accessToken);
+		if (!Objects.equals(values, DELETE)) {
+			throw new NotFoundException(MemberErrorCode.DELETE_FAILED);
+		}
+
+		deleteAssociation(member);
+		replaceWithAnonymous(member);
+		memberRepository.delete(member);
+
+		// oauth2 서비스 연결 끊기
+		oAuth2UnlinkService.unlink(member.getSocialEmail());
+
+		// oauth2 access 토큰 삭제
+		if (redisUtil.getValues("AT(oauth):" + member.getSocialEmail()) != null) {
+			redisUtil.deleteValues("AT(oauth):" + member.getSocialEmail());
+		}
+		return new DeleteMemberResponse(member.getId());
+	}
+
+	private void deleteAssociation(Member member) {
+		creditHistoryRepository.deleteByMember(member);
+		notificationRepository.deleteByMember(member);
+		interactionRepository.deleteByMemberId(member.getId());
+	}
+
+	private void replaceWithAnonymous(Member member) {
+		Member anonymous = memberRepository.findByRole(ANONYMOUS)
+			.orElseThrow(() -> new NotFoundException(MemberErrorCode.NOT_FOUND_MEMBER));
+
+		List<QuestionPost> posts = questionPostRepository.findAllByMember(member);
+		posts.forEach(post -> post.updateMember(anonymous));
+
```

Comment:
```suggestion
@Modifying(flushAutomatically = true, clearAutomatically = true)
@Query("UPDATE QuestionPost q SET q.member = :anonymous WHERE q.member.id = :memberId")
public void updateQuestionPosts(Long memberId, Member anonymous)
```
연관된 엔티티를 모두 찾아서 하나씩 업데이트 해주는 것보다 벌크 업데이트를 해주는게 더 성능이 좋을 것 같아요..!

Reply:
지난 번 벌크 업데이트가 잘 이해되지 않았는데 함께 달아주신 링크를 보니 이해가 잘 되네요.
맨 처음에는 굳이 해야되나 싶었는데, 게시글과 답변 총 합이 수 천개가 되는 계정의 다중 업데이트를 하나씩 하기보다 벌크 업데이트를 통해 한다면 더 좋은 성능이 되겠네요!

좋은 리뷰 감사합니다.